src/object/cli_obj.c

```diff
@@ -3385,6 +3385,16 @@ obj_shard_list_comp_cb(struct shard_auxi_args *shard_auxi,
 
 	shard_arg = container_of(shard_auxi, struct shard_list_args, la_auxi);
 	if (obj_auxi->req_tgts.ort_grp_size == 1) {
+		if (obj_auxi->opc == DAOS_OBJ_RECX_RPC_ENUMERATE &&
+		    shard_arg->la_recxs != NULL) {
+			int i;
+
+			for (i = 0; i < shard_arg->la_nr; i++) {
+				if (shard_arg->la_recxs[i].rx_idx & PARITY_INDICATOR)
+					obj_recx_parity_to_daos(obj_get_oca(obj_auxi->obj),
+								&shard_arg->la_recxs[i]);
+			}
+		}
```

Comment:
please cherrypick the latest version of https://github.com/daos-stack/daos/pull/10531, looks this change is a little bit different with @wangdi1 's closed PR.

Reply:
Thanks for the info.
- [ ] Cherry pick latest version of #10531

Reply:
@liuxuezhao , I have merged the last modification from #10531 . Thank you very much to have pointing me this :)
I am waiting for the CI validation as there is some updates in this patch which seems to be fixed.